### ISSUE TITLE
Add option to only recognize functions that Start-Words-With-Capitals

### ISSUE THREAD

**Issue by anntzer** [PProvost/vim-ps1#32]
Currently, the syntax file recognizes all words-with-dashes as function invocations (`syn match ps1FunctionInvocation /\w\+\(-\w\+\)\+/`), which can yield many false positives, and is also incosistent with the line just above (`syn match ps1FunctionDeclaration /\w\+\(-\w\+\)*/ contained` -- no dash is needed for recognition).

A possibility to decrease the amount of false positives would be to require (perhaps togglable) additionally that the word starts with an uppercase letter (`\u\w*`), which is consistent with MS's style guide (https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/capitalization-conventions).

**Comment by heaths** [comment]
Few scripts I've seen - and almost none I've written - use original casing for cmdlets. However, I'd be open to a change that only recognizes dashed words that start with an approved verb, i.e. output from `get-verb`.

**Comment by RepeatDaily** [comment]
@anntzer I agree. 

@heaths I happen to use original casing nearly always, as do many others, obviously.  Every best practice guide I've seen - including ones from Microsoft - strongly encourage developers to Use-FullCmdletNames; for example [This one](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx#SD02).   At the very least it should be togglable as @anntzer suggested. 

In fact, I was just looking at this the other day as I'd run into the same issue as @anntzer

**Comment by heaths** [comment]
Best practice guidance does not equate to practical usage. Camel casing is not required. If you want to submit a PR for users to *opt into* this behavior, I will consider it.

**Comment by anntzer** [comment]
Here's a patch that only highlights dashed-words that start with an approved verb, as suggested in https://github.com/PProvost/vim-ps1/issues/32#issuecomment-393304994.
```patch
diff --git i/syntax/ps1.vim w/syntax/ps1.vim
index 85150f6..2d67d9e 100644
--- i/syntax/ps1.vim
+++ w/syntax/ps1.vim
@@ -57,16 +57,28 @@ syn match ps1Constant +\$^+
 syn keyword ps1Keyword class define from using var
 
 " Function declarations
-syn keyword ps1Keyword function nextgroup=ps1FunctionDeclaration skipwhite
-syn keyword ps1Keyword filter nextgroup=ps1FunctionDeclaration skipwhite
-syn keyword ps1Keyword workflow nextgroup=ps1FunctionDeclaration skipwhite
-syn keyword ps1Keyword configuration nextgroup=ps1FunctionDeclaration skipwhite
-syn keyword ps1Keyword class nextgroup=ps1FunctionDeclaration skipwhite
-syn keyword ps1Keyword enum nextgroup=ps1FunctionDeclaration skipwhite
-syn match ps1FunctionDeclaration /\w\+\(-\w\+\)*/ contained
+syn keyword ps1Keyword function nextgroup=ps1Function skipwhite
+syn keyword ps1Keyword filter nextgroup=ps1Function skipwhite
+syn keyword ps1Keyword workflow nextgroup=ps1Function skipwhite
+syn keyword ps1Keyword configuration nextgroup=ps1Function skipwhite
+syn keyword ps1Keyword class nextgroup=ps1Function skipwhite
+syn keyword ps1Keyword enum nextgroup=ps1Function skipwhite
 
-" Function invocations
-syn match ps1FunctionInvocation /\w\+\(-\w\+\)\+/
+" Function declarations and invocations
+syn match ps1Function
+    \ /\v(add|clear|close|copy|enter|exit|find|format|get|hide|join|lock|move|new|open|optimize|pop|push|redo|remove|rename|reset|search|select|Set|show|skip|split|step|switch|undo|unlock|watch)(-\w+)+/
+syn match ps1Function
+    \ /\v(connect|disconnect|read|receive|send|write)(-\w+)+/
+syn match ps1Function
+    \ /\v(backup|checkpoint|compare|compress|convert|convertfrom|convertto|dismount|edit|expand|export|group|import|initialize|limit|merge|mount|out|publish|restore|save|sync|unpublish|update)(-\w+)+/
+syn match ps1Function
+    \ /\v(debug|measure|ping|repair|resolve|test|trace)(-\w+)+/
+syn match ps1Function
+    \ /\v(approve|assert|build|complete|confirm|deny|deploy|disable|enable|install|invoke|register|request|restart|resume|start|stop|submit|suspend|uninstall|unregister|wait)(-\w+)+/
+syn match ps1Function
+    \ /\v(block|grant|protect|revoke|unblock|unprotect)(-\w+)+/
+syn match ps1Function
+    \ /\v(use)(-\w+)+/
 
 " Type declarations
 syn match ps1Type /\[[a-z_][a-z0-9_.,\[\]]\+\]/
@@ -162,8 +174,7 @@ if version >= 508 || !exists("did_ps1_syn_inits")
 	HiLink ps1Escape SpecialChar
 	HiLink ps1InterpolationDelimiter Delimiter
 	HiLink ps1Conditional Conditional
-	HiLink ps1FunctionDeclaration Function
-	HiLink ps1FunctionInvocation Function
+	HiLink ps1Function Function
 	HiLink ps1Variable Identifier
 	HiLink ps1Boolean Boolean
 	HiLink ps1Constant Constant
```
To avoid the large repetition I merged psFunctionDeclaration and psFunctionInvocation together, but I can also just dupe the thing.
How does that look to you?  If you agree with the approach I can turn this into a PR (or you can apply the patch directly).

**Comment by heaths** [comment]
Looks good, though I think I prefer the block of adds as a single line per verb group (which, BTW, I love that you did!). Not a big deal, though. Just a little harder to read this way, IMO. Up to you. I would accept a PR either way.

Thank you!

**Comment by anntzer** [comment]
I'm confused, do you prefer one line by verb group (as in the patch) or not? Either way is fine for me.

**Comment by heaths** [comment]
Instead of,
```patch
+syn match ps1Function
+    \ /\v(add|clear|close|copy|enter|exit|find|format|get|hide|join|lock|move|new|open|optimize|pop|push|redo|remove|rename|reset|search|select|Set|show|skip|split|step|switch|undo|unlock|watch)(-\w+)+/
```

Perhaps,
```patch
+syn match ps1Function /\v(add|clear|close|copy|enter|exit|find|format|get|hide|join|lock|move|new|open|optimize|pop|push|redo|remove|rename|reset|search|select|Set|show|skip|split|step|switch|undo|unlock|watch)(-\w+)+/
```

**Comment by anntzer** [comment]
Effectively "closed" by #39.
